staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -46,15 +47,71 @@ type PodResourcesOptions struct {
 	// NonMissingContainerRequests if provided will replace any missing container level requests for the specified resources
 	// with the given values.  If the requests for those resources are explicitly set, even if zero, they will not be modified.
 	NonMissingContainerRequests v1.ResourceList
+	// SkipPodLevelResources controls whether pod-level resources should be skipped
+	// from the calculation. If pod-level resources are not set in PodSpec,
+	// pod-level resources will always be skipped.
+	SkipPodLevelResources bool
 }
 
-// PodRequests computes the pod requests per the PodResourcesOptions supplied. If PodResourcesOptions is nil, then
-// the requests are returned including pod overhead. The computation is part of the API and must be reviewed
-// as an API change.
+var supportedPodLevelResources = sets.New(v1.ResourceCPU, v1.ResourceMemory)
+
+func SupportedPodLevelResources() sets.Set[v1.ResourceName] {
+	return supportedPodLevelResources
+}
+
+// IsSupportedPodLevelResources checks if a given resource is supported by pod-level
+// resource management through the PodLevelResources feature. Returns true if
+// the resource is supported.
+func IsSupportedPodLevelResource(name v1.ResourceName) bool {
+	return supportedPodLevelResources.Has(name)
+}
+
+// IsPodLevelResourcesSet check if PodLevelResources pod-level resources are set.
+// It returns true if either the Requests or Limits maps are non-empty.
+func IsPodLevelResourcesSet(pod *v1.Pod) bool {
+	if pod.Spec.Resources == nil {
+		return false
+	}
+
+	return (len(pod.Spec.Resources.Requests) + len(pod.Spec.Resources.Limits)) > 0
```

Comment:
As above, this will return true if:

```
   resources:
      limits:
        example.com/foobar: "0"
```

Is that OK?

Reply:
Added the check for supported keys; however not checking is quantity is 0

Reply:
That's OK.  I used 0 to make a fake resource actually work.